pkg/provisioner/ironic/inspecthardware_test.go

```diff
@@ -0,0 +1,94 @@
+package ironic
+
+import (
+	"testing"
+	"time"
+
+	"github.com/gophercloud/gophercloud/openstack/baremetal/v1/nodes"
+	"github.com/gophercloud/gophercloud/openstack/baremetalintrospection/v1/introspection"
+	"github.com/metal3-io/baremetal-operator/pkg/bmc"
+	"github.com/metal3-io/baremetal-operator/pkg/provisioner/ironic/clients"
+	"github.com/metal3-io/baremetal-operator/pkg/provisioner/ironic/testserver"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestInspectHardware(t *testing.T) {
+
+	nodeUUID := "33ce8659-7400-4c68-9535-d10766f07a58"
+
+	cases := []struct {
+		name      string
+		ironic    *testserver.IronicMock
+		inspector *testserver.InspectorMock
+
+		expectedDirty        bool
+		expectedRequestAfter int
+		expectedDetailsHost  string
+	}{
+		{
+			name: "inspection-in-progress",
+			ironic: testserver.NewIronic(t).Ready().WithNode(nodes.Node{
+				UUID: nodeUUID,
+			}),
+			inspector: testserver.NewInspector(t).Ready().WithIntrospection(nodeUUID, introspection.Introspection{
+				Finished: false,
+			}),
+
+			expectedDirty:        true,
+			expectedRequestAfter: 15,
+		},
+		{
+			name: "inspection-completed",
+			ironic: testserver.NewIronic(t).Ready().WithNode(nodes.Node{
+				UUID: nodeUUID,
+			}),
+			inspector: testserver.NewInspector(t).Ready().
+				WithIntrospection(nodeUUID, introspection.Introspection{
+					Finished: true,
+				}).
+				WithIntrospectionData(nodeUUID, introspection.Data{
+					Inventory: introspection.InventoryType{
+						Hostname: "node-0",
```

Comment:
Where does the name come from?

Reply:
Just followed the Ironic Inspector API, ie https://docs.openstack.org/ironic-inspector/5.1.0/http-api.html#get-introspection-data, to have roughly a one-to-one relationship